stf/src/sgx.rs

```diff
@@ -200,75 +199,112 @@ impl Stf {
         calls: &mut Vec<OpaqueCall>,
     ) -> Result<(), StfError> {
         let call_hash = blake2_256(&call.encode());
-        ext.execute_with(|| match call.call {
-            TrustedCall::balance_set_balance(root, who, free_balance, reserved_balance) => {
-                Self::ensure_root(root)?;
-                debug!(
-                    "balance_set_balance({:x?}, {}, {})",
-                    who.encode(),
-                    free_balance,
-                    reserved_balance
-                );
-                sgx_runtime::BalancesCall::<Runtime>::set_balance(
-                    MultiAddress::Id(AccountId32::from(who)),
-                    free_balance,
-                    reserved_balance,
-                )
-                .dispatch_bypass_filter(sgx_runtime::Origin::root())
-                .map_err(|_| StfError::Dispatch("balance_set_balance".to_string()))?;
-                Ok(())
-            }
-            TrustedCall::balance_transfer(from, to, value) => {
-                let origin = sgx_runtime::Origin::signed(AccountId32::from(from.clone()));
-                debug!(
-                    "balance_transfer({:x?}, {:x?}, {})",
-                    from.encode(),
-                    to.encode(),
-                    value
-                );
-                if let Some(info) = get_account_info(&from) {
-                    debug!("sender balance is {}", info.data.free);
-                } else {
-                    debug!("sender balance is zero");
+        ext.execute_with(|| {
+            let mut root_account = AccountId::default();
```

Comment:
why do you use the term `root` if it isn't meant as superuser? Why don't you clone the account in-place?

Reply:
or call it `sender` instead?

Reply:
> 
> 
> @haerdib can you confirm that this closes #89?

Yes, I confirm

Reply:
fixed with commit 7f33b2a